components/supervisor/pkg/config/gitpod-config.go

```diff
@@ -87,11 +87,7 @@ func (service *ConfigService) Watch(ctx context.Context) {
 		return
 	}
 
-	_, err := os.Stat(service.location)
-	if os.IsNotExist(err) {
-		service.poll(ctx)
-	}
-	service.watch(ctx)
+	service.poll(ctx)
```

Comment:
maybe rename `pool` to `awaitUntilFileExist` and it should return an error if context is cancelled and finally call `watch` :thinking: